### PR TITLE
[RELAY] Add broadcast_to operator

### PR DESCRIPTION
**Pull request by jroesch**
Add broadcast_to operator for Relay.  cc @siju-samuel @MarisaKirisame @tqchen

### CODE REVIEW

File: python/tvm/relay/op/transform.py [hunk 1]
```diff
@@ -267,6 +267,22 @@ def where(condition, x, y):
     """
     return _make.where(condition, x, y)
 
+def broadcast_to(data, shape):
+    """
+    Return an scalar value array with the same type, but with
+    broadcasted to the broadcast shape.
+     Parameters
```
Comment (MarisaKirisame) [inline python/tvm/relay/op/transform.py]: indent is off

File: python/tvm/relay/op/transform.py [hunk 2]
```diff
@@ -267,6 +267,22 @@ def where(condition, x, y):
...
+    ----------
+    data : relay.Expr
+        The input tensor.
+     broadcast_shape : shape
+        Provide the shape to broadcast to.
+     Returns
```
Comment (MarisaKirisame) [inline python/tvm/relay/op/transform.py]: indent is off

File: src/relay/op/tensor/transform.cc [hunk 3]
```diff
@@ -1084,6 +1084,52 @@ RELAY_REGISTER_OP("collapse_sum_like")
...
+
+Expr MakeBroadCastTo(Expr data, Array<IndexExpr> shape) {
+  static const Op& op = Op::Get("broadcast_to");
+  auto attrs = make_node<InitOpAttrs>();
+  attrs->shape = std::move(shape);
+  return CallNode::make(op, {data}, Attrs(attrs), {});
+}
+
+Array<Tensor> BroadCastToCompute(const Attrs& attrs,
+                                     const Array<Tensor>& inputs,
```
Comment (MarisaKirisame) [inline src/relay/op/tensor/transform.cc]: indent

File: python/tvm/relay/op/transform.py [hunk 4]
```diff
@@ -267,6 +267,22 @@ def where(condition, x, y):
     """
     return _make.where(condition, x, y)
 
+def broadcast_to(data, shape):
+    """
+    Return an scalar value array with the same type, but with
+    broadcasted to the broadcast shape.
```
Comment (MarisaKirisame) [inline python/tvm/relay/op/transform.py]: grammar seems to be off

**Comment by jroesch** [comment]
@tqchen I addressed feedback.

**Comment by tqchen** [comment]
@jroesch given that broadcast_to is a numpy operator(unlike broadcast_to_like), let us set the support level to be 4. Please also update the operator document to include this operator

**Comment by tqchen** [comment]
Thanks, @jroesch , @srkreddy1238 , @MarisaKirisame , this is now merged

**Pull request merged**